test/components/form/table.spec.js

```diff
@@ -1,76 +1,102 @@
 import FormTable from '../../../src/components/form/table.vue';
+import FormRow from '../../../src/components/form/row.vue';
+
 
 import testData from '../../data';
 import { load } from '../../util/http';
 import { mockLogin } from '../../util/session';
+import { mockRouter } from '../../util/router';
+import { mount } from '../../util/lifecycle';
+
+const mountComponent = (showClosed = false) => mount(FormTable, {
+  props: {
+    showClosed,
+    // This is a placeholder sort function. The real one will be
+    // passed through from forms/list.vue
+    sortFunc: (a, b) => a.xmlFormId.localeCompare(b.xmlFormId)
+  },
+  container: {
+    requestData: {
+      forms: testData.extendedForms.sorted(),
+      project: testData.extendedProjects.last()
+    },
+    router: mockRouter('/')
+  }
+});
 
 describe('FormTable', () => {
   describe('number of columns', () => {
     it('shows all columns to an administrator', async () => {
       mockLogin({ role: 'admin' });
-      testData.extendedForms.createPast(1);
+      testData.extendedForms.createPast(1, { state: 'open' });
       const app = await load('/projects/1');
       const table = app.getComponent(FormTable);
       const headers = table.findAll('th').map(th => th.text());
-      headers.should.eql(['Name', 'ID and Version', 'Submissions', 'Actions']);
-      table.findAll('td').length.should.equal(4);
+      headers.should.eql(['Name', 'Review States', 'Latest', 'Total', 'Actions']);
+      table.findAll('td').length.should.equal(7);
     });
 
     it('shows the correct columns to a project viewer', async () => {
       mockLogin({ role: 'none' });
       testData.extendedProjects.createPast(1, { role: 'viewer', forms: 1 });
-      testData.extendedForms.createPast(1);
+      testData.extendedForms.createPast(1, { state: 'open' });
       const app = await load('/projects/1', {}, { deletedForms: false });
       const table = app.getComponent(FormTable);
       const headers = table.findAll('th').map(th => th.text());
-      headers.should.eql(['Name', 'Submissions']);
+      headers.should.eql(['Name', 'Review States', 'Latest', 'Total']);
       const td = table.findAll('td');
-      td.length.should.equal(2);
+      td.length.should.equal(6);
       td[0].classes('name').should.be.true();
```

Comment:
because the only different column for different roles is the "actions" column, these class checks could go away. Or check the class on the last column.